advisor/2016-07-12-preview/swagger/advisor.json

```diff
@@ -0,0 +1,461 @@
+					"Medium",
+					"Low"],
+					"type": "string"
+				},
+				"impactedName": {
+					"type": "string"
+				},
+				"impactedType": {
+					"type": "string"
+				},
+				"lastUpdated": {
+					"format": "date-time",
+					"type": "string"
+				},
+				"shortDescription": {
+					"$ref": "#/definitions/ShortDescription"
+				},
+				"subscriptionId": {
+					"format": "uuid",
+					"type": "string"
+				}
+			}
+		},
+		"ResourceRecomendationAction": {
+			"type": "object",
+			"properties": {
+				"actionType": {
+					"enum": ["Blade",
+					"Document",
+					"PowerShell",
+					"CLI"],
+					"type": "string"
+				},
+				"caption": {
+					"type": "string"
+				},
+				"description": {
+					"type": "string"
+				},
+				"metadata": {
+					"type": "object",
+					"additionalProperties": {
+						"type": "object"
+					}
+				}
+			}
+		},
+		"ShortDescription": {
+			"type": "object",
+			"properties": {
+				"problem": {
+					"type": "string"
+				},
+				"solution": {
+					"type": "string"
+				}
+			}
+		},
+		"CollectionResponse[OperationEntity]": {
+			"type": "object",
+			"properties": {
+				"nextLink": {
+					"type": "string"
+				},
+				"value": {
+					"type": "array",
+					"items": {
+						"$ref": "#/definitions/OperationEntity"
+					}
+				}
+			}
+		},
+		"OperationEntity": {
+			"type": "object",
+			"properties": {
+				"name": {
+					"type": "string"
+				},
+				"display": {
+					"$ref": "#/definitions/OperationDisplayInfo"
+				}
+			}
+		},
+		"OperationDisplayInfo": {
+			"type": "object",
+			"properties": {
+				"description": {
+					"type": "string"
+				},
+				"operation": {
+					"type": "string"
+				},
+				"provider": {
+					"type": "string"
+				},
+				"resource": {
+					"type": "string"
+				}
+			}
+		},
+		"SubscriptionNotificationDefinition": {
+			"type": "object",
+			"properties": {
+				"properties": {
+					"$ref": "#/definitions/SubscriptionNotificationPropertiesDefinition"
+				},
+				"registrationDate": {
+					"type": "string"
+				},
+				"state": {
+					"enum": ["NotDefined",
+					"Registered",
+					"Unregistered",
+					"Warned",
+					"Suspended",
+					"Deleted"],
+					"type": "string"
+				}
+			}
+		},
+		"SubscriptionNotificationPropertiesDefinition": {
+			"type": "object",
+			"properties": {
+				"accountOwner": {
+					"$ref": "#/definitions/SubscriptionAccountOwner"
+				},
+				"locationPlacementId": {
+					"type": "string"
+				},
+				"quotaId": {
+					"type": "string"
+				},
+				"registeredFeatures": {
+					"type": "array",
+					"items": {
+						"$ref": "#/definitions/FeatureDefinition"
+					}
+				},
+				"tenantId": {
+					"type": "string"
+				}
+			}
+		},
+		"SubscriptionAccountOwner": {
+			"type": "object",
+			"properties": {
+				"puid": {
+					"type": "string"
+				}
+			}
+		},
+		"FeatureDefinition": {
+			"type": "object",
+			"properties": {
+				"name": {
+					"type": "string"
+				},
+				"state": {
+					"enum": ["NotSpecified",
```

Comment:
`x-ms-enum` Refer https://github.com/Azure/autorest/tree/master/docs/extensions#x-ms-enum and apply as necessary to model SDK correctly.